### PR TITLE
Add email for Stripe free trial ending event

### PR DESCRIPTION
**Pull request by bryce-fitzsimons**
### Features and Changes
- Adds 2 new emails as part of a common jinja template for free trial ending, triggered by Stripe's `customer.subscription.trial_will_end` event.
  - trial ending: no card on file
  - trial ending: card on file will be billed

### CODE REVIEW

File: packages/back-end/src/controllers/stripe.ts [hunk 1]
```diff
@@ -235,6 +236,29 @@ export async function postWebhook(req: Request, res: Response) {
...
+      case "customer.subscription.trial_will_end": {
+        const responseSubscription = event.data
+          .object as Stripe.Response<Stripe.Subscription>;
+        if (responseSubscription) {
+          const ret = await updateSubscriptionInDb(responseSubscription);
+          if (ret) {
+            const { organization, subscription, hasPaymentMethod } = ret;
+            const endDate = subscription.trial_end
+              ? new Date(subscription.trial_end * 1000)
+              : new Date(new Date().getTime() + 3 * 24 * 60 * 60 * 1000);
```
Comment (jdorn) [inline packages/back-end/src/controllers/stripe.ts:249]: If there's no trial_end for the subscription, something went wrong, and we should skip sending the email (and log an error).

File: packages/back-end/src/services/email.ts [hunk 2]
```diff
@@ -217,3 +218,60 @@ export async function sendPendingMemberApprovalEmail(
...
+    to: email,
+    text,
+  });
+}
+
+// todo: use shared package date utils:
+function daysLeft(date: string | Date): number {
+  return differenceInDays(getValidDate(date), new Date());
+}
+function getValidDate(
```
Comment (jdorn) [inline packages/back-end/src/services/email.ts:264]: Shared package is merged to main now

Comment (bryce-fitzsimons) [reply packages/back-end/src/services/email.ts]: updated this and the shared package

File: packages/shared/src/dates.ts [hunk 3]
```diff
@@ -1,3 +1,33 @@
+import format from "date-fns/format";
+import formatDistance from "date-fns/formatDistance";
+import differenceInDays from "date-fns/differenceInDays";
+import { addMonths } from "date-fns";
```
Comment (jdorn) [inline packages/shared/src/dates.ts:4]: Switch the style of this import to match the others (better tree shaking)

**Review by jdorn — APPROVED** [review]

**Review by jdorn — APPROVED** [review]

**Pull request merged**